theano/tensor/extra_ops.py

```diff
@@ -608,7 +608,16 @@ def repeat(x, repeats, axis=None):
 
     .. versionadded:: 0.6
     """
-    return RepeatOp(axis=axis)(x, repeats)
+    ndim  = x.ndim
+    shape = [x.shape[i] for i in xrange(ndim)]
+    if axis is None:
+        rep_tile = [1 for i in xrange(ndim)] + [repeats] 
+        z = tensor.flatten(tensor.tile(x.reshape(shape+[1]), rep_tile))
```

Comment:
It would be cleaner to have `x[numpy.newaxis]` rather than `x.reshape(shape + [1])`, since we are only adding a new dimension, not really reshaping.